src/stores/events/EventsStore.ts

```diff
@@ -181,41 +185,115 @@ export default class EventsStore {
 		}
 	};
 
-	private eventTreeAC: AbortController | null = null;
-
 	@action
-	private fetchEventTree = async () => {
-		if (this.eventTreeAC) {
-			this.eventTreeAC.abort();
-		}
-		this.eventTreeAC = new AbortController();
-		this.eventTreeStatusCode = null;
+	private fetchEventTree = () => {
+		return new Promise<void>((res, rej) => {
+			this.eventTreeStatusCode = null;
+			this.selectedNode = null;
+			this.isLoadingRootEvents = true;
+			this.eventTree = [];
 
-		this.selectedNode = null;
-		this.isLoadingRootEvents = true;
+			try {
+				const eventSource = this.api.sse.getEventSource({
```

Comment:
sse channel should be member of eventstore and it should be close upon applying filter etc